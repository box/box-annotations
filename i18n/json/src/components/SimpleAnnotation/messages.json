[
  {
    "id": "ba.anonymousUserName",
    "description": "Placeholder when the current annotation's user information is unknown",
    "defaultMessage": "Some User"
  },
  {
    "id": "ba.whoHighlighted",
    "description": "Label for who highlighted the annotated text",
    "defaultMessage": "{name} highlighted"
  },
  {
    "id": "ba.whoDrew",
    "description": "Label for who drew the drawing annotation",
    "defaultMessage": "{name} drew"
  },
  {
    "id": "ba.whoAnnotated",
    "description": "Label for who left the annotation",
    "defaultMessage": "{name} annotated"
  }
]