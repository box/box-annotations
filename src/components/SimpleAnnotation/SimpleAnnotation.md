### Description
Component to represent a simple drawing/highlight annotation dialog

### Examples
```jsx
const SimpleAnnotationExamples = require('examples').SimpleAnnotationExamples;

<SimpleAnnotationExamples />
```
