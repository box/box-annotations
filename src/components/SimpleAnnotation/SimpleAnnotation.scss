.ba {
    .ba-annotation-simple {
        align-items: center;
        display: flex;
    }

    .ba-annotation-label {
        line-height: 29px;
        margin-bottom: 3px;
    }
}
