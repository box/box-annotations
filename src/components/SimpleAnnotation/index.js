// @flow
export { default } from './SimpleAnnotation';
