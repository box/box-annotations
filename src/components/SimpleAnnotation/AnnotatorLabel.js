// @flow
import React from 'react';
import { injectIntl } from 'react-intl';

import CommentText from '../../../third-party/components/Comment/CommentText';

import { TYPES } from '../../constants';
import { isHighlightAnnotation } from '../../util';

import messages from './messages';

// import './AnnotatorLabel.scss';

type Props = {
    id: string,
    type: AnnotationType,
    user?: User,
    intl: any
};

class AnnotatorLabel extends React.PureComponent<Props> {
    getAnnotationLabel() {
        const { type, user, intl } = this.props;
        const anonymousUserName = intl.formatMessage(messages.anonymousUserName);
        const annotatorName = !!user && !!user.name ? user.name : anonymousUserName;

        if (isHighlightAnnotation(type)) {
            return intl.formatMessage(messages.whoHighlighted, { name: annotatorName });
        }

        if (type === TYPES.draw) {
            return intl.formatMessage(messages.whoDrew, { name: annotatorName });
        }

        return intl.formatMessage(messages.whoAnnotated, { name: annotatorName });
    }

    render() {
        const { id } = this.props;
        const annotatorLabelMessage = this.getAnnotationLabel();

        return (
            <span className='ba-annotation-label'>
                <CommentText id={id} tagged_message={annotatorLabelMessage} translationEnabled={false} />
            </span>
        );
    }
}

export default injectIntl(AnnotatorLabel);
