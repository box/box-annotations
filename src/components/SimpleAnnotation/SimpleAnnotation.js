// @flow
import React from 'react';

import Internationalize from '../Internationalize';

// import './SimpleAnnotation.scss';
import AnnotatorLabel from './AnnotatorLabel';
import ActionControls from '../ActionControls';

type Props = {
    id: string,
    type: AnnotationType,
    canDelete: boolean,
    canAnnotate: boolean,
    onCreate: Function,
    onCommentClick: Function,
    onDelete: Function,
    isPending: boolean,
    currentUser?: User,
    language?: string,
    messages?: StringMap,
    intl: any
};

const SimpleAnnotation = ({
    id,
    type,
    canDelete,
    canAnnotate,
    onCreate,
    onCommentClick,
    onDelete,
    isPending,
    currentUser,
    language,
    messages: intlMessages
}: Props) => {
    return (
        <Internationalize language={language} messages={intlMessages}>
            <div className='ba-annotation-simple'>
                {!isPending && <AnnotatorLabel id={id} type={type} currentUser={currentUser} />}
                <ActionControls
                    type={type}
                    canDelete={canDelete}
                    canAnnotate={canAnnotate}
                    onCreate={onCreate}
                    onCommentClick={onCommentClick}
                    onDelete={onDelete}
                />
            </div>
        </Internationalize>
    );
};

export default SimpleAnnotation;
