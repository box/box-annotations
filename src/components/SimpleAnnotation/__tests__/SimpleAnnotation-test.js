import * as React from 'react';
import { shallow } from 'enzyme';

import SimpleAnnotation from '../SimpleAnnotation';

const onDelete = jest.fn();
const onCreate = jest.fn();
const onCommentClick = jest.fn();

const USER = {
    type: 'user',
    id: '789ghi',
    name: 'Mia Thermopolis',
    email: 'princess@genovia.gov'
};

describe('components/SimpleAnnotation', () => {
    const render = (props = {}) =>
        shallow(
            <SimpleAnnotation
                id='123'
                canAnnotate={true}
                canDelete={true}
                onCreate={onCreate}
                onCommentClick={onCommentClick}
                onDelete={onDelete}
                isPending={false}
                {...props}
            />
        );

    test('should correctly render a pending highlight dialog', () => {
        const wrapper = render({ type: 'highlight', isPending: true });
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render a pending drawing dialog', () => {
        const wrapper = render({ type: 'draw', isPending: true });
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render a highlight dialog', () => {
        const wrapper = render({ type: 'highlight', isPending: true });
        expect(wrapper).toMatchSnapshot();
    });

    test('should correctly render a drawing dialog', () => {
        const wrapper = render({ type: 'draw', isPending: true });
        expect(wrapper).toMatchSnapshot();
    });
});
