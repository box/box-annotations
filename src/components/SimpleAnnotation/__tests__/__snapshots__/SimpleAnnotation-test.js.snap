// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SimpleAnnotation should correctly render a drawing dialog 1`] = `
<Internationalize>
  <div
    className="ba-annotation-simple"
  >
    <ActionControls
      canAnnotate={true}
      canDelete={true}
      onCommentClick={[MockFunction]}
      onCreate={[MockFunction]}
      onDelete={[MockFunction]}
      type="draw"
    />
  </div>
</Internationalize>
`;

exports[`components/SimpleAnnotation should correctly render a highlight dialog 1`] = `
<Internationalize>
  <div
    className="ba-annotation-simple"
  >
    <ActionControls
      canAnnotate={true}
      canDelete={true}
      onCommentClick={[MockFunction]}
      onCreate={[MockFunction]}
      onDelete={[MockFunction]}
      type="highlight"
    />
  </div>
</Internationalize>
`;

exports[`components/SimpleAnnotation should correctly render a pending drawing dialog 1`] = `
<Internationalize>
  <div
    className="ba-annotation-simple"
  >
    <ActionControls
      canAnnotate={true}
      canDelete={true}
      onCommentClick={[MockFunction]}
      onCreate={[MockFunction]}
      onDelete={[MockFunction]}
      type="draw"
    />
  </div>
</Internationalize>
`;

exports[`components/SimpleAnnotation should correctly render a pending highlight dialog 1`] = `
<Internationalize>
  <div
    className="ba-annotation-simple"
  >
    <ActionControls
      canAnnotate={true}
      canDelete={true}
      onCommentClick={[MockFunction]}
      onCreate={[MockFunction]}
      onDelete={[MockFunction]}
      type="highlight"
    />
  </div>
</Internationalize>
`;
